src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -0,0 +1,101 @@
+package reposense.parser;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+import java.util.Optional;
+
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.impl.Arguments;
+import net.sourceforge.argparse4j.impl.action.HelpArgumentAction;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.Namespace;
+import reposense.exception.ParseException;
+import reposense.frontend.CliArguments;
+
+
+/**
+ * Verifies and parses a string-formatted date to a {@code CliArguments} object.
+ */
+public class ArgsParser {
+    private static final String PROGRAM_USAGE = "java -jar RepoSense.jar";
+    private static final String PROGRAM_DESCRIPTION =
+            "RepoSense is a contribution analysis tool for Git repositories.";
+    private static final String MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE =
+            "\"Since Date\" cannot be later than \"Until Date\"";
+
+    private static ArgumentParser getArgumentParser() {
+        ArgumentParser parser = ArgumentParsers
+                .newFor(PROGRAM_USAGE)
+                .addHelp(false)
+                .build()
+                .description(PROGRAM_DESCRIPTION);
+
+        parser.addArgument("-h", "--help")
+                .help("Show help message.")
+                .action(new HelpArgumentAction());
+
+        parser.addArgument("-config")
+                .required(true)
+                .type(Arguments.fileType().verifyExists().verifyIsFile().verifyCanRead())
+                .metavar("PATH")
+                .help("The path to the CSV config file to read.");
+
+        parser.addArgument("-output")
+                .metavar("PATH")
+                .type(Arguments.fileType().verifyExists().verifyIsDirectory().verifyCanWrite())
+                .setDefault(new File("."))
+                .help("The path to the dashboard generated. "
+                        + "If not provided, it will be generated in the current directory.");
+
+        parser.addArgument("-since")
+                .metavar("dd/MM/yyyy")
+                .type(new DateArgumentType())
+                .setDefault(Optional.empty())
+                .help("The date to start filtering.");
+
+        parser.addArgument("-until")
+                .metavar("dd/MM/yyyy")
+                .type(new DateArgumentType())
+                .setDefault(Optional.empty())
+                .help("The date to stop filtering.");
+
+        return parser;
+    }
+
+    /**
+     * Parses the given string arguments to a {@code CliArguments} object.
+     * @throws ParseException if the given string fails to parse to a CliArguments object.
```

Comment:
`@throws ParseException if the given string arguments fails to parse to a {@code CliArguments} object`